setup.py

```diff
@@ -369,29 +370,47 @@ def finalize_options(self):
 
     def run(self):
         from PyQt6 import uic
+
         _translate_re = (
-            re.compile(
+            (re.compile(r'(\s+_translate = QtCore\.QCoreApplication\.translate)'), r''),
+            (re.compile(
                 r'QtGui\.QApplication.translate\(.*?, (.*?), None, '
-                r'QtGui\.QApplication\.UnicodeUTF8\)'),
-            re.compile(
-                r'\b_translate\(.*?, (.*?)(?:, None)?\)')
+                r'QtGui\.QApplication\.UnicodeUTF8\)'), r'_(\1)'),
+            (re.compile(r'\b_translate\(.*?, (.*?)(?:, None)?\)'), r'_(\1)'),
         )
 
         def compile_ui(uifile, pyfile):
-            log.info("compiling %s -> %s", uifile, pyfile)
             tmp = StringIO()
+            log.info("compiling %s -> %s", uifile, pyfile)
             uic.compileUi(uifile, tmp)
             source = tmp.getvalue()
-            rc = re.compile(r'\n\n#.*?(?=\n\n)', re.MULTILINE | re.DOTALL)
-            comment = ("\n\n# Automatically generated - don't edit.\n"
-                       "# Use `python setup.py %s` to update it."
-                       % _get_option_name(self))
-            for r in list(_translate_re):
-                source = r.sub(r'_(\1)', source)
-                source = rc.sub(comment, source)
-            f = open(pyfile, "w")
-            f.write(source)
-            f.close()
+
+            # replace QT translations stuff by ours
+            for matcher, replacement in _translate_re:
+                source = matcher.sub(replacement, source)
+
+            # replace headers
+            rc = re.compile(r'\n# WARNING.*?(?=\nclass )', re.MULTILINE | re.DOTALL)
+
+            command = _get_option_name(self)
+            new_header = f"""
+# Automatically generated - do not edit.
+# Use `python setup.py {command}` to update it.
+
+from PyQt6 import (
+    QtCore,
+    QtGui,
+    QtWidgets,
+)
```

Comment:
If I understand correctly this now hard codes the imports for Qt to the above three, while previously the import would be kept as generated. There is a potential this will break if a UI file would cause other imports for some reason. Not sure if that's a thing, so.

Reply:
Yes, I thought about it, but for now, only those 3 imports are done everywhere, so that's not an issue. Though the change in the generated comment actually caused an issue (fixed in this PR), so I guess we just need to adapt if something arises, at least any change will be easy to spot since it will break quite obviously. Having imports ordered isn't really important, since those are ignored due to https://github.com/metabrainz/picard/blob/ebfd011a9981ec23845fb5213a2b70782b415a3e/pyproject.toml#L18
I'd say if any issue arises, it will be easy to fix anyway.